extensions/sql-migration/src/wizard/skuRecommendationPage.ts

```diff
@@ -845,16 +846,29 @@ export class SKURecommendationPage extends MigrationWizardPage {
 	}
 
 	private _onTdeConfigClosed(): Thenable<void> {
-		const tdeMsg = (this.migrationStateModel.tdeMigrationConfig.isTdeMigrationMethodAdsConfirmed()) ? constants.TDE_WIZARD_MSG_TDE : constants.TDE_WIZARD_MSG_MANUAL;
+		const tdeMsg = (this.migrationStateModel.tdeMigrationConfig.getAppliedConfigDialogSetting() === ConfigDialogSetting.ExportCertificates) ? constants.TDE_WIZARD_MSG_TDE : constants.TDE_WIZARD_MSG_MANUAL;
 		this._tdedatabaseSelectedHelperText.value = constants.TDE_MSG_DATABASES_SELECTED(this.migrationStateModel.tdeMigrationConfig.getTdeEnabledDatabasesCount(), tdeMsg);
 
-		const tdeTelemetryAction = (this.migrationStateModel.tdeMigrationConfig.isTdeMigrationMethodAdsConfirmed()) ? TelemetryAction.TdeConfigurationUseADS : TelemetryAction.TdeConfigurationIgnoreADS;
+		let tdeTelemetryAction: TelemetryAction;
+
+		switch (this.migrationStateModel.tdeMigrationConfig.getAppliedConfigDialogSetting() === ConfigDialogSetting.ExportCertificates) {
+			case true:
+				tdeTelemetryAction = TelemetryAction.TdeConfigurationUseADS;
+				break;
+			case false:
+				tdeTelemetryAction = TelemetryAction.TdeConfigurationAlreadyMigrated;
+				break;
+			default:
+				tdeTelemetryAction = TelemetryAction.TdeConfigurationCancelled;
+				break;
```

Comment:
I don't think default will be hit here, would it make sense to switch on the value returned instead?

Reply:
You're right, good call!